Feature/series details

Mohamed-Ashraf-2210 opened the pull request:
get series details from remote 


<img width="1366" height="768" alt="Screenshot (37)" src="https://github.com/user-attachments/assets/399746bc-eecf-4ae5-91b6-e1490ae3bf04" />

<img width="848" height="754" alt="image" src="https://github.com/user-attachments/assets/fcc5f262-e871-4bdc-9594-e277ee493933" />

Ashraf-El-Fallah committed:
initialize cast , last seasons , recommended series , series details and  top reviews use cases for series details screen

Ashraf-El-Fallah committed:
remove person from series details entities

Ashraf-El-Fallah committed:
finish domain and data implement for series details

Ashraf-El-Fallah committed:
Merge remote-tracking branch 'origin/feature/series-details-domain' into feature/series-details-domain

# Conflicts:
#	domain/series/src/main/java/com/giraffe/series/usecase/GetLastSeasonsUseCase.kt
#	domain/series/src/main/java/com/giraffe/series/usecase/GetSeriesDetailsUseCase.kt

Mohamed-Ashraf-2210 committed:
refactor(entities): Clean up SeriesDetails entity and genres handling

- Remove unused `unrequitedVariable` from SeriesDetails entity
- Replace SeriesGenre with `GenreInfo` sealed class to unify:
  - Genre ID (for series search result)
  - Full Genre (for series details result)
- Remove incomplete functions

Mohamed-Ashraf-2210 committed:
add unit tests for two use cases

Mohamed-Ashraf-2210 committed:
remove Series Details entity "duplication" and using series entity

mohamedebrahem13 commented on datasource/remote/series/src/main/java/com/giraffe/series/TmdbSeriesApiRemoteDataSource.kt:
```diff
@@ -46,4 +48,13 @@ class TmdbSeriesApiRemoteDataSource(
     }
+
+    override suspend fun getSeriesDetails(seriesId: Int): SeriesDetailsResponse {
+        val endpoint = "tv/$seriesId"
```
create constant for the end point

mohamedebrahem13 commented on datasource/remote/series/src/main/java/com/giraffe/series/TmdbSeriesApiRemoteDataSource.kt:
```diff
@@ -46,4 +48,13 @@ class TmdbSeriesApiRemoteDataSource(
+        baseRequest.endpoint(endpoint)
+            .method(HttpMethod.Get)
+        val result = requestBuilder.request(baseRequest).body<SeriesDetailsResponse>()
+        Log.d("result", result.toString())
```
remove the log

mohamedebrahem13 commented on domain/series/src/main/java/com/giraffe/series/usecase/GetLastSeasonsUseCase.kt:
```diff
@@ -0,0 +1,10 @@
+
+class GetLastSeasonsUseCase(private val seriesRepository: SeriesRepository) {
+    suspend operator fun invoke(seriesId: Int): List<Season> {
+        return seriesRepository.getSeriesDetails(seriesId).seasons
```
here return the seasons direct form the repo

mohamedebrahem13 commented on domain/series/src/test/kotlin/com/giraffe/series/usecase/GetLastSeasonsUseCaseTest.kt:
```diff
@@ -0,0 +1,43 @@
+        val result = getLastSeasonsUseCase(seriesId)
+
+        // Then
+        assertThat(expectedResult).isEqualTo(result)
```
use truth from google for assertion

mohamedebrahem13 commented on domain/series/src/test/kotlin/com/giraffe/series/usecase/GetSeriesDetailsUseCaseTest.kt:
```diff
@@ -0,0 +1,39 @@
+        val result = getSeriesDetailsUseCase(seriesId)
+
+        // Then
+        assertThat(expectedResult).isEqualTo(result)
```
use truth from google for assertion

mohamedebrahem13 commented on repository/series/src/main/java/com/giraffe/series/utils/safeCall.kt:
```diff
@@ -30,6 +31,7 @@ suspend inline fun <reified T> safeCall(
     return try {
         block()
     } catch (e: Throwable) {
+        Log.d("Throw", e.toString())
```
remove this logs

Mohamed-Ashraf-2210 committed:
replace get last season use case return from series.season to list of season direct

skaik-mo commented on repository/series/src/main/java/com/giraffe/series/mapper/mapper.kt:
```diff
@@ -17,7 +25,8 @@ fun CachedSeriesDto.toEntity(
 }
-fun CachedSeasonDto.toEntity(): Season {
+
+fun CachedSeasonDto.toSeriesEntity(): Season {
```
Why change `toEntity` to `toSeriesEntity`? `toEntity` is clearer and aligns with common mapping conventions from `DTO` to `Domain` entities.

Ashraf-El-Fallah committed:
finish reviews for series in data and domain

Ashraf-El-Fallah committed:
Merge remote-tracking branch 'origin/feature/series-details' into feature/series-details

# Conflicts:
#	domain/series/src/main/java/com/giraffe/series/repository/SeriesRepository.kt
#	repository/series/src/main/java/com/giraffe/series/SeriesRepositoryImpl.kt
#	repository/series/src/main/java/com/giraffe/series/mapper/mapper.kt

Ashraf-El-Fallah committed:
create specific naming for mapping

poula22 committed:
feature: -get recommended series based on series id
- create GetRecommendedSeriesUseCase
- add GetRecommendedSeries function to repository
- add GetRecommendedSeries to remote datasource
- add GetRecommendedSeriesUseCase dependency to koin module
- add unit test with test coverage

poula22 committed:
Merge remote-tracking branch 'origin/feature/series-details' into feature/series-details-recommended-series

# Conflicts:
#	datasource/remote/series/src/main/java/com/giraffe/series/TmdbSeriesApiRemoteDataSource.kt
#	domain/series/src/main/java/com/giraffe/series/repository/SeriesRepository.kt
#	repository/series/src/main/java/com/giraffe/series/SeriesRepositoryImpl.kt
#	repository/series/src/main/java/com/giraffe/series/datasource/remote/SeriesRemoteDataSource.kt

poula22 committed:
merge with series details and solve conflict

poula22 committed:
fix bug in recommended series url string typo

poula22 committed:
- fix bug in getSeriesRecommendations page key has constant value.
- replace assertEquals with asserThat in tests

Mohamed-Ashraf-2210 committed:
Merge remote-tracking branch 'origin/develop' into feature/series-details

mohannadahmed00 committed:
Merge pull request #188 from Rome-Squad/feature/series-details-recommended-series

Feature/get recommended series domain and remote data source

Mohamed-Ashraf-2210 committed:
fix series response error

Mohamed-Ashraf-2210 committed:
remove comment unnecessary from tests

Mohamed-Ashraf-2210 committed:
create test for get series reviews use case

Mohamed-Ashraf-2210 committed:
add onclick function

Mohamed-Ashraf-2210 committed:
Merge remote-tracking branch 'origin/develop' into feature/series-details

# Conflicts:
#	datasource/remote/user/build.gradle.kts
